docs/dev/Lua API.rst

```diff
@@ -5766,82 +5766,129 @@ The names of the functions are also available as the keys of the
 building-hacks
 ==============
 
-This plugin overwrites some methods in workshop df class so that mechanical workshops are possible. Although
-plugin export a function it's recommended to use lua decorated function.
+This plugin overwrites some methods in workshop df class to change and extend workshop
+functionality. See exported function list for details.
 
 .. contents::
   :local:
 
 Functions
 ---------
 
-``registerBuilding(table)`` where table must contain name, as a workshop raw name, the rest are optional:
+* ``setOwnableBuilding(workshop_type, enable)``
 
-    :name:
-        custom workshop id e.g. ``SOAPMAKER``
+  Set workshop to be included in zones (such as bedroom or inn).
 
-        .. note:: this is the only mandatory field.
+  :workshop_type:   custom workshop string id e.g. ``SOAPMAKER`` or numeric id
+  :enable:   boolean value to enable or disable this functionality
 
-    :fix_impassible:
-        if true make impassable tiles impassable to liquids too
-    :consume:
-        how much machine power is needed to work.
-        Disables reactions if not supplied enough and ``needs_power==1``
-    :produce:
-        how much machine power is produced.
-    :needs_power:
-        if produced in network < consumed stop working, default true
-    :gears:
-        a table or ``{x=?,y=?}`` of connection points for machines.
-    :action:
-        a table of number (how much ticks to skip) and a function which
-        gets called on shop update
-    :animate:
-        a table of frames which can be a table of:
+* ``fixImpassible(workshop_type, enable)``
 
-        a. tables of 4 numbers ``{tile,fore,back,bright}`` OR
-        b. empty table (tile not modified) OR
-        c. ``{x=<number> y=<number> + 4 numbers like in first case}``,
-           this generates full frame useful for animations that change little (1-2 tiles)
+  Set workshop non walkable tiles to also block liquids (i.e. water and magma).
 
-    :canBeRoomSubset:
-        a flag if this building can be counted in room. 1 means it can, 0 means it can't and -1 default building behaviour
-    :auto_gears:
-        a flag that automatically fills up gears and animations. It looks over the building definition for gear icons and maps them.
+  :workshop_type:   custom workshop string id e.g. ``SOAPMAKER`` or numeric id
+  :enable:   boolean value to enable or disable this functionality
 
-    Animate table also might contain:
+* ``setMachineInfo(workshop_type, needs_power, power_consumed, power_produced, connection_points)``
```

Comment:
do we need a separate `needs_power` parameter? Can this be combined with `power_consumed`? e.g. if `power_consumed` is > 0, then the building needs power? Or is there a case where the building still works even if it is without needed power?

maybe what I'm not understanding is: how do the values specified here interact with `setPower`? would it make sense to remove the `power_consumed` and `power_produced` parameters from this function and require that `setPower` be called separately?

Reply:
`needs_power` is a bit misnamed argument. It make it function like a vanilla millstone: you can't use it while it's unpowered.

Reply:
as for `power_consumed/produced` yeah it probably could be removed and require `setPower` to be called. Will make the api cleaner

Reply:
Ah i just tried removing `power_consumed/produced` thingy and remembered: `setPower` sets power for one specific building and the `setMachineInfo` sets for all buildings of that type. Though it's not intended to be called more than once.